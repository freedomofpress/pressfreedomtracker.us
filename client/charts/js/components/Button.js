import React from "react";

const textStyle = {
	fontFamily: "Helvetica Neue",
	fontWeight: "500",
	fontSize: "14px",
};

export function Button({ label, selected, selectable = true, onClick }) {
	const [hovered, setHovered] = React.useState(false);
	return (
		<button
			style={{
				padding: 8,
				border: "none",
				marginBottom: 3,
				outline: "solid 3px black",
				backgroundColor:
					selected || (hovered && selectable) ? "black" : "white",
				color:
					selected || (hovered && selectable)
						? "white"
						: selectable
						? "black"
						: "grey",
				cursor: selectable ? "pointer" : "default",
			}}
			onClick={() => {
				if (selectable) {
					onClick();
				}
			}}
			onMouseEnter={() => {
				setHovered(true);
			}}
			onMouseLeave={() => {
				setHovered(false);
			}}
		>
			<p
				style={{
					margin: 0,
					fontFamily: textStyle.fontFamily,
					fontSize: textStyle.fontSize,
					fontWeight: textStyle.fontWeight,
				}}
			>
				{label}
			</p>
		</button>
	);
}
