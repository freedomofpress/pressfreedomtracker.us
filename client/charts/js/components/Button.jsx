import React from 'react'
import Tooltip from './Tooltip'

const textStyle = {
	fontFamily: 'var(--font-base)',
	fontWeight: '500',
	fontSize: '14px',
}

export default function Button({ label, selected, selectable = true, onClick, tooltipIfUnselectable = null }) {
	const [hovered, setHovered] = React.useState(false)
	const [tooltipPosition, setTooltipPosition] = React.useState({ x: 0, y: 0 })

	const updateTooltipPosition = (MouseEvent) => {
		setTooltipPosition({ x: MouseEvent.clientX, y: MouseEvent.clientY })
	}

	return (
		<>
		{hovered && !selectable && tooltipIfUnselectable && (
			<Tooltip
				x={tooltipPosition.x}
				y={tooltipPosition.y}
				content={
					<div style={{ fontFamily: 'var(--font-base)', fontSize: 12, fontWeight: 500 }}>
						{tooltipIfUnselectable}
					</div>
				}
			/>
		)}
		<button
			style={{
				padding: 8,
				border: 'none',
				marginBottom: 3,
				outline: 'solid 3px black',
				backgroundColor: selected || (hovered && selectable) ? 'black' : 'white',
				color: selected || (hovered && selectable) ? 'white' : selectable ? 'black' : 'grey',
				cursor: selectable ? 'pointer' : 'default',
				minWidth: 50,
			}}
			onMouseMove={updateTooltipPosition}
			onClick={() => {
				if (selectable) {
					onClick()
				}
			}}
			onMouseEnter={(e) => {
				updateTooltipPosition(e)
				setHovered(true)
			}}
			onMouseLeave={() => {
				setHovered(false)
			}}
		>
			<p
				style={{
					margin: 0,
					fontFamily: textStyle.fontFamily,
					fontSize: textStyle.fontSize,
					fontWeight: textStyle.fontWeight,
				}}
			>
				{label}
			</p>
		</button>
	</>
	)
}
