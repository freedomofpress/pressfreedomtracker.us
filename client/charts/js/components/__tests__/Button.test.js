import React from 'react'
import ReactTestRenderer from 'react-test-renderer'
import Button from '../Button'

test('renders Button with mocked data', () => {
	expect(ReactTestRenderer.create(
		<Button
			label="test"
			selected={false}
			onClick={() => {}}
		/>
	)).toMatchSnapshot();
});
