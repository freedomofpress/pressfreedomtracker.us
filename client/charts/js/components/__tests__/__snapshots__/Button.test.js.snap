// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Button with mocked data 1`] = `
<React.Fragment>
  <button
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseMove={[Function]}
    style={
      {
        "backgroundColor": "white",
        "border": "none",
        "color": "black",
        "cursor": "pointer",
        "marginBottom": 3,
        "minWidth": 50,
        "outline": "solid 3px black",
        "padding": 8,
      }
    }
  >
    <p
      style={
        {
          "fontFamily": "var(--font-base)",
          "fontSize": "14px",
          "fontWeight": "500",
          "margin": 0,
        }
      }
    >
      test
    </p>
  </button>
</React.Fragment>
`;
